pkg/gameserverallocations/allocator.go

```diff
@@ -273,10 +275,10 @@ func (c *Allocator) allocateFromLocalCluster(ctx context.Context, gsa *allocatio
 		return nil, err
 	}
 
-	switch err {
-	case ErrNoGameServer:
+	switch {
+	case goErrors.Is(err, ErrNoGameServer), goErrors.Is(err, ErrGameServerUpdateConflict):
```

Comment:
```suggestion
	case ErrGameServerUpdateConflict:
```

same below, since we don't really use `goErrors` except to Join errors together below, and this is only needed on goErrors.Wrap(), and `pkg/errors` doesn't support `errors.Unwrap() error`

Maybe one day we'll do a big migration project? 😁